### PR TITLE
Move test CI from CircleCI to Github Actions

### PR DESCRIPTION
**Pull request by adrw**
Given recent CircleCI security issues, this aligns Backfila repo with most other CashApp repos which are already on Github Actions. A follow up PR will remove the CircleCI config but for now we will dual-run CI.

### CODE REVIEW

File: bin/jaotc [hunk 1]
```diff
@@ -0,0 +1 @@
+.openjdk@11.pkg
```
Comment (shellderp) [inline bin/jaotc:1]: are these bin files intentional?

Comment (adrw) [reply bin/jaotc:1]: Yes, they're part of the Hermit config and meant to be checked in. For example in Misk: https://github.com/cashapp/misk/tree/master/bin

Comment (shellderp) [reply bin/jaotc:1]: ok it's not obvious at a glance I guess, this works better than a file listing deps?

File: bin/activate-hermit [hunk 2]
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+# This file must be used with "source bin/activate-hermit" from bash or zsh.
```
Comment (shellderp) [inline bin/activate-hermit:2]: so I run this to set up my environment after pulling this change?

Comment (shellderp) [reply bin/activate-hermit:2]: the home README.md should be updated to say how you get dev env set up with hermit

Comment (adrw) [reply bin/activate-hermit:2]: https://github.com/cashapp/backfila/pull/291

File: bin/jaotc (hunk 1 above)
Comment (adrw) [reply bin/jaotc]: Yes, I initially was hesitant but over the past year it's made toolchain management a lot simpler across internal and external repos.

File: .github/workflows/Test.yml [hunk 3]
```diff
@@ -0,0 +1,76 @@
+name: Test
```
Comment (adrw) [inline .github/workflows/Test.yml:1]: File copied from Misk with Backfila DB docker line added

File: build.gradle.kts [hunk 4]
```diff
@@ -115,3 +115,8 @@ subprojects {
     }
   }
 }
+
+// Disable the Gradle wrapper if Gradle is managed by Hermit
```
Comment (adrw) [inline build.gradle.kts:119]: Missed adding this in the Hermit PR #291

**Review by danieloh0714 — APPROVED** [review]

**Review by r3mariano — APPROVED** [review]

**Review by shellderp — APPROVED** [review]
oke

**Pull request merged**